plugins/contentful/src/plugin.ts

```diff
@@ -0,0 +1,143 @@
+import { registerDataPlugin } from '@builder.io/data-plugin-tools';
+import pkg from '../package.json';
+import contentful from 'contentful';
+import qs from 'qs';
+
+const pluginId = pkg.name;
+const metaFields = ['environment', 'space', 'revision', 'type'];
+
+registerDataPlugin(
+  {
+    id: pluginId,
+    name: 'Contentful',
+    icon:
+      'https://cdn.builder.io/api/v1/image/assets%2FYJIGb4i01jvw0SRdL5Bt%2Fd6097cd40fef4b94b18a3e0c4c53584d',
+    settings: [
+      {
+        name: 'spaceId',
+        type: 'string',
+        required: true,
+        helperText:
+          'Get your space ID, from your contentful space settings > API Keys https://www.contentful.com/developers/docs/references/authentication/',
+      },
+      {
+        name: 'accessToken',
+        type: 'string',
+        required: true,
+        helperText:
+          'Get your access token, from your contentful space settings > API Keys https://www.contentful.com/developers/docs/references/authentication/',
+      },
+    ],
+    ctaText: `Connect your Contentful space`,
+  },
+  // Observable map of the settings configured above
+  async settings => {
+    const spaceId = settings.get('spaceId')?.trim();
+    const accessToken = settings.get('accessToken')?.trim();
+    const client = await contentful.createClient({
+      space: spaceId,
+      accessToken,
+    });
+    return {
+      async getResourceTypes() {
+        const contentTypes = await client.getContentTypes();
+        return contentTypes.items.map(type => ({
+          name: type.name,
+          id: type.sys.id,
+          canPickEntries: true,
+          inputs: () => {
+            const acceptableFields = type.fields.filter(field =>
+              ['Text', 'Boolean', 'Number', 'Symbol'].includes(field.type)
+            );
+            const fields: any = [
```

Comment:
I think you can use the `Input` type from our SDK no?

```suggestion
            const fields: Input[] = [
```